arbos/internal_tx.go

```diff
@@ -55,6 +56,13 @@ func ApplyInternalTxUpdate(tx *types.ArbitrumInternalTx, state *arbosState.Arbos
 			return err
 		}
 
+		var prevHash common.Hash
+		if evm.Context.BlockNumber.Sign() > 0 {
+			prevHash = evm.Context.GetHash(evm.Context.BlockNumber.Uint64() - 1)
+		}
+		if state.ArbOSVersion() >= params.ArbosVersion_40 && evm.ChainConfig().IsArbitrum() {
```

Comment:
I wonder if this should be IsPrauge()?

Also, can we add a comment referring to EIP-2935 so we know why we're making this call?

Reply:
For IsPrauge we dont have easy access to L1 block time.
Also IsPrauge is very geth specific call, in nitro codebase we mostly check the arbos_version only, like https://github.com/OffchainLabs/nitro/blob/master/arbos/tx_processor.go#L790

Added the comment regarding EIP-2935